src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,154 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head = null;
+    private Node<T> tail = null;
+
     @Override
     public void add(T value) {
+        Node<T> node;
+        node = new Node<>(null, value,null);
+        if (head == null) {
+            head = tail = node;
+        } else {
+            tail.next = node;
+            node.prev = tail;
+        }
+        tail = node;
```

Comment:
The tail should be updated only in the `else` block. As it stands, you're updating the tail twice when the head is null.